README.md

```diff
@@ -20,9 +20,53 @@ This chart deploys a fully functioning instance of the [Fn](https://github.com/f
 
 
 ```bash
-helm init
+helm init --upgrade
 ```
 
+## Preparing chart values
+
+### Minimum configuration
+
+In order to get a working deployment please pay attention to what you have in your chart values.
+[Here](fn/values.yaml) you may find bare minimum chart values configuration necessary to deploy a working Fn cluster.
+
+### Exposing Fn services
+
+#### Ingress controller
+
+If you are installing Fn behind an ingress controller you'd need to have only 1 DNS sub-domain that will stand for an ingress controller IP resolve.
+
+Please pay attention, an ingress controller works as a proxy, it means that you can use ingress IP address as an HTTP proxy:
+
+```
+curl -x http://<ingress-controller-endpoint>:80 api.fn.internal 
+{"goto":"https://github.com/fnproject/fn","hello":"world!"}
+```
+
+
+#### LoadBalancer
+
+In order to use native way to expose Fn services you'd need to modify Fn API, Fn runner LB and UI service definitions:
+
+ - at `fn_api` node values, modify `fn_api.service.type` from `ClusterIP` to `LoadBalancer`
+ - at `fn_lb_runner` node values, modify `fn_lb_runner.service.type` from `ClusterIP` to `LoadBalancer`
+ - at `ui` node values, modify `ui.service.type` from `ClusterIP` to `LoadBalancer`
+
+
+#### DNS names
+
+In Fn deployment with LoadBalancer service types, you'd need 3 DNS names:
+
+ - one for an API service (i.e., `api.fn.mydomain.com`)
+ - one for runner LB service (i.e., `lb.fn.mydomain.com`)
+ - one for UI service (i.e., `ui.fn.mydomain.com`)
+
+With the successful deployment you'd have 3 public IP addresses for each of services.
+However, an IP address for API and LB services would be identical since they are expose as a single service.
+So, you'd have 2 IP address, but 3 DNS names.
+
+But still, the best way for exposing services is an **ingress controller**.
```

Comment:
Please keep in mind the best way for exposing services is an ingress controller.